src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -249,23 +248,28 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
   def generateJesOutputs(backendCall: BackendCall): Seq[JesOutput] = {
     val wdlFileOutputs = backendCall.call.task.outputs flatMap { taskOutput =>
       taskOutput.expression.evaluateFiles(backendCall.lookupFunction, new NoFunctions, taskOutput.wdlType) match {
-        case Success(wdlFiles) => wdlFiles map gcsPathToLocal map (_.valueString)
+        case Success(wdlFiles) => wdlFiles map gcsPathToLocal
         case Failure(ex) =>
           logger.warn(s"${makeTag(backendCall)} Could not evaluate $taskOutput: ${ex.getMessage}")
           Seq.empty[String]
       }
     }
-    wdlFileOutputs.distinct map { filePath =>
-      JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/$filePath", localFilePathFromRelativePath(filePath))
+
+    // Create the mappings. GLOB mappings require special treatment (i.e. stick everything matching the glob in a folder)
+    wdlFileOutputs.distinct map {
+      file => file match {
+        case WdlFile(filePath, GlobOrNot.NotGlob) => JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/$filePath", localFilePathFromRelativePath(filePath))
+        case WdlFile(filePath, GlobOrNot.YesGlob) => JesOutput(makeSafeJesReferenceName(filePath), s"${backendCall.callGcsPath}/globbed-${DigestionUtil.md5Sum(filePath)}/", localFilePathFromRelativePath(filePath))
+      }
     }
   }
 
   /**
    * If the desired reference name is too long, we don't want to break JES or risk collisions by arbitrary truncation. So,
-   * make it a nice random string.
+   * just use a hash. We only do this when needed to give better traceability in the normal case.
    */
   private def makeSafeJesReferenceName(referenceName: String) = {
-    if (referenceName.length <= 127) referenceName else UUID.randomUUID().toString
+    if (referenceName.length <= 127) referenceName else DigestionUtil.md5Sum(referenceName)
```

Comment:
`Util`s seemed to be frowned upon in Scala, perhaps a renamed version of this object could provide an implicit class enhancer for string?

```scala
implicit class StringMD5Digest(val string: String) extends AnyVal {
  def md5: String = ...
}
```